app/bundles/EmailBundle/Tests/Model/EmailModelBuildUrlTest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\EmailBundle\Tests\Model;
+
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\EmailBundle\Model\EmailModel;
+
+class EmailModelBuildUrlTest extends MauticMysqlTestCase
+{
+    protected function setUp(): void
+    {
+        $this->configParams['site_url'] = 'https://foo.bar.com';
+        parent::setUp();
+    }
+
+    public function testSiteUrlAlwaysTakesPrecedenceWhenBuildingUrls(): void
+    {
+        /** @var EmailModel $emailModel */
+        $emailModel = self::$container->get('mautic.email.model.email');
```

Comment:
Just for next time, it's better to use the `assert` instead of type hints.
```suggestion
        $emailModel = self::$container->get('mautic.email.model.email');
        \assert($emailModel instanceof EmailModel)
```